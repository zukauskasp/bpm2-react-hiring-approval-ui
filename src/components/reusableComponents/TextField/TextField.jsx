import React from "react";
import './TextField.scss';
import PropTypes from 'prop-types';
import Tooltip from "../Tooltip/Tooltip";

const TextField = ({ textarea, readonly, name, label, required, type, value, placeholder, onChange, rows, errorMessage, ...otherProps }) => {

    if (textarea) {
        return (
            <div className="form-group">
                <label className={`${readonly ? 'readOnlyLabel' : null}`} htmlFor={name}>{label} {required ? " *" : null}</label>
                <textarea type={type} name={name} className={`textArea ${readonly ? 'readOnly' : null}`} value={value} placeholder={placeholder} onChange={onChange} rows={rows} readOnly={readonly} {...otherProps} />
                {errorMessage ? <Tooltip text={errorMessage} messageType='error' /> : null}
            </div>
        );
    } else {
        return (
            <div className="form-group">
                {label ? <label className={`${readonly ? 'readOnlyLabel' : null}`} htmlFor={name}>{label} {required ? " *" : null}</label> : null}
                <input type={type} name={name} className={`form-control  ${readonly ? 'readOnly' : null}`} value={value} placeholder={placeholder} onChange={onChange} readOnly={readonly} {...otherProps} />
                {errorMessage ? <Tooltip text={errorMessage} messageType='error' /> : null}
            </div>
        );
    }
}

TextField.propTypes = {
    errorMessage: PropTypes.string,
    onChange: PropTypes.func.isRequired,
    label: PropTypes.string,
    required: PropTypes.bool,
    readonly: PropTypes.bool,
    textarea: PropTypes.bool,
    name: PropTypes.string,
    type: PropTypes.oneOf(['text', 'number', 'password', 'time']),
    value: PropTypes.string,
    placeholder: PropTypes.string,
    rows: PropTypes.number,
}

TextField.defaultProps = {
    errorMessage: "",
    label: "",
    rows: 3,
    type: "text"
}

export default TextField;