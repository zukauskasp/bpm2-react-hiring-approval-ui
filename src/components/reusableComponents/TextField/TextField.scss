.form-group {
    margin-bottom: 24px;
    position: relative;
    width: 280px;

    .form-control {
        font-family: 'Danske Text v2';
        color: #003755;
        height: 48px;
        display: block;
        width: 100%;
        padding: .375rem .75rem;
        font-size: 16px;
        line-height: 48px;
        background-color: #fff;
        background-clip: padding-box;
        border: 1px solid #ced4da;
        border-radius: .25rem;
        transition: border-color .15s ease-in-out, box-shadow .15s ease-in-out;

        &:focus {
            border-color: rgb(0, 159, 218);
            outline: 0 none;
        }

        ::placeholder {
            color: #7F9BAA;
        }
    }

    .textArea {
        font-family: 'Danske Text v2';
        display: block;
        width: 100%;
        padding: .375rem .75rem;
        font-size: 16px;
        height: 96px;
        line-height: 1.5;
        color: #495057;
        background-color: #fff;
        background-clip: padding-box;
        border: 1px solid #ced4da;
        border-radius: .25rem;
        transition: border-color .15s ease-in-out, box-shadow .15s ease-in-out;

        &:focus {
            border-color: rgb(0, 159, 218);
            outline: 0 none;
        }

        ::placeholder {
            color: #7F9BAA;
        }
    }

    .readOnly {
        color: #7F9BAA;
        cursor: no-drop;

        &:focus {
            border: 1px solid #ced4da;
        }

        ::placeholder {
            color: #7F9BAA;
        }
    }

    label {
        font-family: 'Danske Text v2';
        display: inline-block;
        margin-bottom: 8px;
        font-size: 14px;
        color: #003755;
    }

    .readOnlyLabel {
        color: #7F9BAA;
        cursor: default;
    }
}