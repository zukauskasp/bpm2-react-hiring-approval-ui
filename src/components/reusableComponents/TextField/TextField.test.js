import React from 'react';
import { shallow, mount } from 'enzyme';

import TextField from './TextField';

describe('TextField', () => {

    let wrapper;

    const props = {
        required: false,
        label: "enter number",
        onChange: () => { },
        errorMessage: "Error message",
        rows: 3,
        type: "number"
    }

    beforeEach(() => {
        wrapper = shallow(<TextField {...props} />);
    })

    afterAll(() => {
        mount.cleanUp();
      });

    it('expect to match latest snapshot', () => {
        expect(shallow(<TextField {...props} />).debug()).toMatchSnapshot();
    });

    it('expect to render component', (() => {
        expect(wrapper.length).toEqual(1);
    }));

    it('expect to render correct props', () => {
        expect(mount(<TextField {...props} />).props()).toEqual(props);
    });

    it('expect to add correct classes', () => {
        const readOnly = shallow(<TextField {...props} readonly={true} />);

        expect(readOnly.find('input').prop('className').indexOf('readOnly') > -1).toEqual(true);
    });

    it('expect to simulate event', () => {
        const mockCallBack = jest.fn();

        shallow(<TextField {...props} onChange={mockCallBack} />).find("input").simulate('change');
        expect(mockCallBack.mock.calls.length).toEqual(1);
    });

    it('expect to render textarea', () => {
        const textArea = shallow(<TextField {...props} textarea={true} />);

        expect(textArea.find('textarea').length).toEqual(1);
    });

    it('expect to render error', () => {
        const textArea = shallow(<TextField {...props} />);

        expect(textArea.find('Message').length).toEqual(1);
    });

});