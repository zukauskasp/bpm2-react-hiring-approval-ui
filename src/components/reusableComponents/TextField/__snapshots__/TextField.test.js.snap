// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField expect to match latest snapshot 1`] = `
"<div className=\\"form-group\\">
  <label className=\\"null\\" htmlFor={[undefined]}>
    enter number
     
  </label>
  <input type=\\"number\\" name={[undefined]} className=\\"form-control  null\\" value={[undefined]} placeholder={[undefined]} onChange={[Function: onChange]} readOnly={[undefined]} />
  <Message text=\\"Error message\\" messageType=\\"error\\" position=\\"top-left\\" />
</div>"
`;
